### PR TITLE
💡 [patch] Repo maintenance: deps upgrades

### PR DESCRIPTION
**Pull request by ronjouch**
The usual:

1. Relock
2. Major upgrade, see individual commits for proof migration guides were followed

## How was this tested?

Tests & linter pass

### CODE REVIEW

File: src/lib/RedisCache.ts [hunk 1]
```diff
@@ -60,7 +60,6 @@ export class RedisCache extends CacheInstance {
       // This prevents get/setValue calls from hanging if there is no active connection
       enableOfflineQueue: false,
     });
-    this.redisClient.scan
```
Comment (ronjouch) [inline src/lib/RedisCache.ts:63]: Must have been a cat walking on keyboard :wink: . Introduced in [this repo maintenance](https://github.com/unitoio/cachette/pull/68/files#diff-b4f3c69471b4f85d48c8ef8bf675571d2f15fda598050a34dff4e467f6187f56R60). Thankfully, was not doing anything.

File: eslint.config.mjs
Comment (ronjouch) [inline eslint.config.mjs:1]: Reminder to reviewers: Cachette is public, no revealing internal sikrit world domination plans here in comments. 🙏

File: eslint.config.mjs [hunk 2]
```diff
@@ -0,0 +1,29 @@
+import eslint from '@eslint/js';
```
Comment (ronjouch) [inline eslint.config.mjs:1]: **Reminder to reviewers: Cachette is public**, no revealing internal sikrit world domination plans here in comments. 🙏

**Review by gregsavoie — APPROVED** [review]

File: src/lib/RedisCache.ts (hunk 1 above)
Comment (ianfdk) [reply src/lib/RedisCache.ts:63]: 😳 close call

**Review by ianfdk — APPROVED** [review]

**Pull request merged**